### ISSUE TITLE
Uncaught TypeError: Cannot read property 'path' of undefined

### ISSUE THREAD

**Issue by nakedcity** [hex13/atom-lupa#18]
[Enter steps to reproduce below:]
1. ...
2. ...

**Atom Version**: 1.7.4
**System**: Mac OS X 10.11.4
**Thrown From**: [atom-lupa](https://github.com/hex13/atom-lupa) package, v0.5.48
### Stack Trace

Uncaught TypeError: Cannot read property 'path' of undefined

```
At /Users/camiloazula/.atom/packages/atom-lupa/node_modules/react/lib/ReactErrorUtils.js:52

TypeError: Cannot read property 'path' of undefined
    at /Users/camiloazula/.atom/packages/atom-lupa/lib/lupaMiddleware.js:31:60
    at /Users/camiloazula/.atom/packages/atom-lupa/lib/middleware/domMiddleware.js:22:9
    at Object.dispatch (/Users/camiloazula/.atom/packages/atom-lupa/node_modules/redux-thunk/lib/index.js:14:16)
    at _react2.default.createElement._react2.default.createElement.onClick (/Users/camiloazula/.atom/packages/atom-lupa/lib/components/Lupa.js:253:34)
    at Object.invokeGuardedCallback (/Users/camiloazula/.atom/packages/atom-lupa/node_modules/react/lib/ReactErrorUtils.js:26:12)
    at executeDispatch (/Users/camiloazula/.atom/packages/atom-lupa/node_modules/react/lib/EventPluginUtils.js:87:21)
    at Object.executeDispatchesInOrder (/Users/camiloazula/.atom/packages/atom-lupa/node_modules/react/lib/EventPluginUtils.js:110:5)
    at executeDispatchesAndRelease (/Users/camiloazula/.atom/packages/atom-lupa/node_modules/react/lib/EventPluginHub.js:42:22)
    at executeDispatchesAndReleaseTopLevel (/Users/camiloazula/.atom/packages/atom-lupa/node_modules/react/lib/EventPluginHub.js:53:10)
    at Array.forEach (native)
    at forEachAccumulated (/Users/camiloazula/.atom/packages/atom-lupa/node_modules/react/lib/forEachAccumulated.js:24:9)
    at Object.EventPluginHub.processEventQueue (/Users/camiloazula/.atom/packages/atom-lupa/node_modules/react/lib/EventPluginHub.js:215:7)
    at runEventQueueInBatch (/Users/camiloazula/.atom/packages/atom-lupa/node_modules/react/lib/ReactEventEmitterMixin.js:18:18)
    at Object.ReactEventEmitterMixin.handleTopLevel [as _handleTopLevel] (/Users/camiloazula/.atom/packages/atom-lupa/node_modules/react/lib/ReactEventEmitterMixin.js:29:5)
    at handleTopLevelImpl (/Users/camiloazula/.atom/packages/atom-lupa/node_modules/react/lib/ReactEventListener.js:73:24)
    at ReactDefaultBatchingStrategyTransaction.Mixin.perform (/Users/camiloazula/.atom/packages/atom-lupa/node_modules/react/lib/Transaction.js:136:20)
    at Object.ReactDefaultBatchingStrategy.batchedUpdates (/Users/camiloazula/.atom/packages/atom-lupa/node_modules/react/lib/ReactDefaultBatchingStrategy.js:63:19)
    at Object.batchedUpdates (/Users/camiloazula/.atom/packages/atom-lupa/node_modules/react/lib/ReactUpdates.js:98:20)
    at ReactEventListener.dispatchEvent (/Users/camiloazula/.atom/packages/atom-lupa/node_modules/react/lib/ReactEventListener.js:150:20)
```
### Commands

```
     -0:20.2.0 atom-lupa:toggle (ol.tree-view.full-menu.list-tree.has-collapsable-children.focusable-panel)
     -0:11 tree-view:toggle (input.lupa-search.native-key-bindings)
```
### Config

``` json
{}
```
### Installed Packages

``` coffee
# User
atom-lupa, v0.5.48
atom-ternjs, v0.14.1
autocomplete-modules, v1.4.1
autocomplete-paths, v1.0.2
language-babel, v2.23.2
language-postcss, v1.2.0
linter, v1.11.4
linter-eslint, v7.2.2
linter-jsonlint, v1.2.5
linter-stylelint, v2.10.0
local-history, v3.3.0
merge-conflicts, v1.4.3
standard-formatter, v2.0.0

# Dev
No dev packages
```

**Comment by hex13** [comment]
Have you had focused on some file before activating Lupa? Because it was bug which triggered when user was for example in Settings and then triggered Index Project (related to: https://github.com/hex13/atom-lupa/issues/14 )

Bug was fixed but it's sort of reggresion due to refactoring ( need for test: #19 )

`TypeError: Cannot read property 'path' of undefined
    at /Users/camiloazula/.atom/packages/atom-lupa/lib/lupaMiddleware.js:31:60`

(Unfortunately Lupa in current version needs user to open some file before indexing. I know this is a little cumbersome. In future versions I will work to improve that. But for now this is how it works. 
But nevertheless it shouldn't break or throw exceptions, so I've fixed for now this exception.
